third_party/blink/renderer/core/layout/ng/ng_out_of_flow_layout_part.cc

```diff
@@ -1587,11 +1589,17 @@ bool NGOutOfFlowLayoutPart::TryCalculateOffset(
   //  - The candidate has an inline container (instead of the default
   //    containing-block).
   // Note: Only check for cache results if this is our first layout pass.
+  recordreplay::Assert(
+      "[RUN-1239] NGOutOfFlowLayoutPart::TryCalculateOffset A %d",
+      is_first_run, test_if_margin_box_fits, allow_first_tier_oof_cache_, node_info.inline_container);
   if (is_first_run && !test_if_margin_box_fits && allow_first_tier_oof_cache_ &&
       !node_info.inline_container) {
-    if (const NGLayoutResult* cached_result =
-            node_info.node.CachedLayoutResultForOutOfFlowPositioned(
-                container_content_size_in_candidate_writing_mode)) {
+    const NGLayoutResult* cached_result =
+        node_info.node.CachedLayoutResultForOutOfFlowPositioned(
+            container_content_size_in_candidate_writing_mode);
+    recordreplay::Assert(
+        "[RUN-1239] NGOutOfFlowLayoutPart::TryCalculateOffset B %d", !!cached_result);
```

Comment:
In general I'd prefer it if we avoided making edits to the upstream chromium source and just added asserts when possible (it isn't always possible though, e.g. if we want to assert a function's return value).  In this case we could have asserts in the true and false branches that accomplished the same thing.  This makes it easier to read the diffs vs upstream and reason about the possible effects the record/replay changes can have, and will make rebases easier as well.

Reply:
Fixed. Turns out we only need one of the two branches anyway!